pkg/tecdsa/dkg/preparams.go

```diff
@@ -0,0 +1,84 @@
+package dkg
+
+import (
+	"time"
+
+	"github.com/binance-chain/tss-lib/ecdsa/keygen"
+	"github.com/ipfs/go-log"
+	"github.com/keep-network/keep-core/pkg/miner"
+)
+
+// PreParams represents tECDSA DKG pre-parameters that were not yet consumed
+// by DKG protocol execution.
+type PreParams struct {
+	data keygen.LocalPreParams
+}
+
+// NewPreParams constructs a new instance of tECDSA DKG pre-parameters based on
+// the generated numbers.
+func NewPreParams(data keygen.LocalPreParams) *PreParams {
+	return &PreParams{data}
+}
+
+// tssPreParamsPool is a pool holding TSS pre parameters. It autogenerates
+// entries up to the pool size. When an entry is pulled from the pool it
+// will generate a new entry.
+type tssPreParamsPool struct {
+	*miner.ParameterPool[keygen.LocalPreParams]
```

Comment:
Shouldn't we build the pool around `dkg.PreParams` instead of `keygen.LocalPreParams`? The reason behind is that the former implements `Marshal` and `Unmarshal` so it can be persisted while the latter cannot. Also, if the storage implementation is implemented somewhere outside, tss-lib internal type will leak outside of the `pkg/tecdsa`.

Reply:
Indeed! 45d5df162d4bd69546b4b98d729cbaa489739e6b